internal/runbits/runtimemsg.go

```diff
@@ -3,61 +3,97 @@ package runbits
 // Progress bar design
 //
 import (
+	"context"
 	"fmt"
+	"strconv"
+	"time"
 
-	"github.com/go-openapi/strfmt"
+	"github.com/thoas/go-funk"
+	"github.com/vbauerster/mpb/v6"
 
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
 	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/runbits/progressbar"
 	"github.com/ActiveState/cli/pkg/platform/runtime/artifact"
-	"github.com/ActiveState/cli/pkg/platform/runtime/setup"
+	"github.com/ActiveState/cli/pkg/platform/runtime/setup/events"
 )
 
 type RuntimeMessageHandler struct {
 	out output.Outputer
 }
 
 func NewRuntimeMessageHandler(out output.Outputer) *RuntimeMessageHandler {
```

Comment:
Looking at the code again in my IDE I now see that you're mainly just using the message handler to pass a channel and then do most of your logic via the channel. This pretty much nillifies the purpose of the message handler. Please commit to one or the other. (Use callbacks or use channels, not both). My preference goes to callbacks because it's much easier to work with. Write dumb code.